src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+    private static final int ADD_EVENT = 6;
+
+    public static Task[] Tasks = new Task[100];
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printGreeting();
+        loopCommands();
+        sayGoodbye();
+    }
+
+    public static int getCommand(String line) {
+        String[] subStrings = line.split(" ");
+        if (subStrings[0].equalsIgnoreCase("bye")) {
+            return EXIT;
+        }
+        if (subStrings[0].equalsIgnoreCase("done")) {
+            return MARK;
+        }
+        if (subStrings[0].equalsIgnoreCase("list")) {
+            return LIST;
+        }
+        return ADD;
+    }
+    public static int getItemType(String line) {
+        if(line.equalsIgnoreCase("todo")) {
+            return ADD_TODO;
+        }
+        if(line.equalsIgnoreCase("deadline")) {
+            return ADD_DEADLINE;
+        }
+        if(line.equalsIgnoreCase("event")) {
+            return ADD_EVENT;
+        }
+        return ERR_WRONG_FORMAT;
+    }
+    /**
+     * Loops through all features available.
+     * Returns if user inputs bye or number of tasks exceeds 100
+     */
+    public static void loopCommands() {
+        while (true) {
+            if (Task.totalTasks > 100) {
+                System.out.println("Maximum capacity reached");
+                return;
+            }
+            Scanner in = new Scanner(System.in);
+            String line;
+            while (in.hasNextLine()) {
+                line = in.nextLine();
+                int command = getCommand(line);
+                // No fallthrough required
+                switch (command) {
+                // If user wants to mark a task as done
+                case MARK:
+                    markAsDone(line);
+                    break;
+                // If user wants to list all tasks
+                case LIST:
+                    listItems();
+                    break;
+                case ADD:
+                    // If user wants to add an item
+                    addItem(line);
+                    break;
+                case EXIT:
+                    return;
+                }
+            }
+        }
+    }
+    /**
+     * Marks a task as done
+     * Checks for out of bounds access and presence of numerical value
+     * @param line user input
+     */
+    public static void markAsDone(String line) {
+        int listNum = checkValidDone(line);
+        if(listNum == INVALID_DONE) {
+            printError(ERR_OUT_OF_BOUNDS);
+        } else {
+            listNum -= 1;
+            Tasks[listNum].setDone();
+            System.out.println(Tasks[listNum].getName() + " set to done. You're well smart innit?");
```

Comment:
Could put these strings into a constant to make reading and future editing of such strings easier (e.g. final static MESSAGE_SET_TO_DONE). Applies to the other message strings as well.